core/scmi-server/sub.mk

```diff
@@ -0,0 +1,45 @@
+# SCMI server library is built from SCP-firmware source tree.
+# The firmware is made of the framework, a product and modules
+# from either the generic path (subdir module/) or form the
+# product path (subdir product/*/module/).
+
+scmi-server-scp-path = $(CFG_SCP_FIRMWARE)
+scmi-server-product := $(CFG_SCMI_SERVER_PRODUCT)
+scmi-server-out-path = $(out-dir)/$(libdir)
+
+srcs-y += scmi_server.c
+incdirs-y += include
+
+incdirs_ext-y += $(scmi-server-scp-path)/arch/none/optee/include
+
+scp-firmware-output = $(scmi-server-out-path)/build/product/optee-$(scmi-server-product)/fw/libscmi-fw-all.a
+
+libdeps += $(scp-firmware-output)
+
+cleanfiles += $(scp-firmware-output)
+
+libdeps += out/arm/core-lib/libutils/libutils.a
+
+scmi-server-flags-y = -DSCP_FIRMWARE_SOURCE_DIR:PATH=optee-$(scmi-server-product)/fw
+scmi-server-flags-$(CFG_TEE_CORE_DEBUG) += -DSCP_LOG_LEVEL="TRACE"
+scmi-server-flags-y += -DDISABLE_CPPCHECK=1
+scmi-server-flags-y += -DCFG_NUM_THREADS=$(CFG_NUM_THREADS)
+scmi-server-flags-y += -DSCP_OPTEE_DIR:PATH=$(CURDIR)
+
+scmi-server-flags-y += -DCFG_CROSS_COMPILE=$(lastword $(CROSS_COMPILE_core))
+
+CFLAGS_OPTEE = $(cflags$(sm))
+CFLAGS_OPTEE += $(subst -include $(conf-file),,$(cflags$(sm)))
+CFLAGS_OPTEE += $(cflags-plat-scmi-server-y)
+scmi-server-flags-y += -DCFG_CFLAGS_OPTEE="$(CFLAGS_OPTEE)"
```

Comment:
`$(cflags$(sm))` can change even after this `sub.mk` has been processed.
Have you verified that `$(CFLAGS_OPTEE)` is updated as needed if/when that happens?

Reply:
Latest change do evaluate `$(cflagscore)` when target is executed, taking into account changes to its value after this _sub.mk_ file is parsed.

Reply:
Thanks for checking.